tools/gulp/tasks/docs.ts

```diff
@@ -177,3 +180,197 @@ function createTagNameAliaser(classPrefix: string) {
+        selectorName: ${res.selectorName ? ("'" + res.selectorName + "'") : 'undefined'}
+    },
+    `;
+  };
+
+  // create the examples component list
+  const list = (res: ExampleMetadata): string => {
+    const components = [res.component];
+
+    if (res.additionalComponents) {
+      components.push(...res.additionalComponents);
+    }
+    return `${components.join(',')},
+    `;
+  };
+
+  return `
+  import {NgModule} from '@angular/core';
+  import {FormsModule, ReactiveFormsModule} from '@angular/forms';
+  import {CommonModule} from '@angular/common';
+  import {
+    MdAutocompleteModule, MdButtonModule, MdButtonToggleModule, MdPaginatorModule,
+    MdCardModule, MdCheckboxModule, MdChipsModule, MdDatepickerModule,
+    MdDialogModule, MdGridListModule, MdIconModule, MdInputModule,
+    MdListModule, MdMenuModule, MdProgressBarModule, MdProgressSpinnerModule,
+    MdRadioModule, MdSelectModule, MdSidenavModule, MdSliderModule,
+    MdSlideToggleModule, MdSnackBarModule, MdTabsModule, MdToolbarModule, MdTooltipModule
+  } from '@angular/material';
+
+  export interface LiveExample {
+    title: string;
+    component: any;
+    additionalFiles?: string[];
+    selectorName?: string;
+  }
+
+  ${results.map(r => imports(r)).join('').trim()}
+
+  export const EXAMPLE_COMPONENTS = {
+    ${results.map(r => examples(r)).join('').trim()}
+  };
+
+  @NgModule({
+    exports: [
+      MdAutocompleteModule,
+      MdButtonModule,
+      MdButtonToggleModule,
+      MdCardModule,
+      MdCheckboxModule,
+      MdChipsModule,
+      MdDatepickerModule,
+      MdDialogModule,
+      MdGridListModule,
+      MdIconModule,
+      MdInputModule,
+      MdListModule,
+      MdMenuModule,
+      MdProgressBarModule,
+      MdProgressSpinnerModule,
+      MdRadioModule,
+      MdSelectModule,
+      MdSlideToggleModule,
+      MdSliderModule,
+      MdSidenavModule,
+      MdSnackBarModule,
+      MdTabsModule,
+      MdToolbarModule,
+      MdTooltipModule,
+      MdPaginatorModule
+    ]
+  })
+  export class ExampleMaterialModule {}
+
+  export const EXAMPLE_LIST = [
+    ${results.map(r => list(r)).join('').trim()}
+  ];
+
+  @NgModule({
+    declarations: EXAMPLE_LIST,
+    entryComponents: EXAMPLE_LIST,
+    imports: [
+      ExampleMaterialModule,
+      FormsModule,
+      ReactiveFormsModule,
+      CommonModule,
+    ]
+  })
+  export class ExampleModule { }
+`;
+};
+
+/**
+ * Creates the examples module and metadata
+ */
+task('build-examples-module', (cb) => {
+  const results: ExampleMetadata[] = [];
+
+  glob('src/material-examples/**/*.ts', (err, matches) => {
+    if (err) {
+      throw err;
+    }
+
+    for (const match of matches) {
+      const src = fs.readFileSync(match, 'utf-8');
+      const filename = match
+        .replace('src/material-examples/', './')
+        .replace('.ts', '');
+
+      // get the decorator tags, not all required
+      const titleMatch = src.match(/@title.*/g);
```

Comment:
Yeah most of the `additionalFiles` and `additionalComponents` could be detected using the TypeScript AST. 

The only thing that is problematic would be the "demo" title for a given demo (because it wouldn't fit into the metadata)

Although the title could be part of the JSDoc of a component.

```ts
/**
 * @title Test Demo
 */
@Component({
  ...
}
export class MyTestExample {}
```